### ISSUE TITLE
tls timeout

### ISSUE THREAD

**Issue by celesteking** [baudehlo/Haraka#921]
Getting: [ERROR] [F9118AD1-991C-4C89-AD2F-7EAF93F3B555] [tls] timeout

Happening on cipher mismatch.
1. Test yourself with:  openssl s_client -connect  localhost:2525  -starttls smtp  -cipher ECDHE-ECDSA-AES128-SHA
2. wait for plugin.timeout time

Wondering if that's an error or not. And if it's okay, why is it logged with ERROR level then?

**Comment by msimerson** [comment]
````
$ time  openssl s_client -connect mail.theartfarm.com:587 -starttls smtp -cipher ECDHE-ECDSA-AES128-SHA
CONNECTED(00000003)
140735212979024:error:140790E5:SSL routines:ssl23_write:ssl handshake failure:s23_lib.c:177:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 245 bytes and written 205 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
---

real	0m3.327s
user	0m0.004s
sys	0m0.002s
````

The 3 second duration is because of karma delays and a very large upload that's currently congesting my very crappy Comcast upload bandwidth.

The results of running `openssl s_client -connect localhost:25 -starttls smtp -cipher ECDHE-ECDSA-AES128-SHA` locally on my server are comparable. I am running the latest patch-of-the-month version (1.0.1l) of openssl.

Haraka logs:
````
Apr 12 12:13:35 node haraka[39750]: [NOTICE] [95E05881-08D9-4253-9D96-452D86B52E55] [core] connect ip=50.159.99.59 port=33243 local_ip=127.0.0.30 local_port=587
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [connect.p0f] os="Mac OS X " link_type="Ethernet or modem" distance=12 total_conn=4 shared_ip=N
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [connect.fcrdns] ip=50.159.99.59 rdns="c-50-159-99-59.hsd1.wa.comcast.net" rdns_len=1 fcrdns="c-50-159-99-59.hsd1.wa.comcast.net" fcrdns_len=1 other_ips_len=0 invalid_tlds=0 generic_rdns=true
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] applied results.connect.fcrdns.fcrdns:1
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] applied results.connect.fcrdns.fail:-1
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [access] pass:connect:any
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [connect.geoip] asn: 7922, org: Comcast Cable Communications, Inc.
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [connect.geoip] NA, US, WA, Seattle, 2699km
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [dnsbl] fail:pbl.spamhaus.org
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] applied results.dnsbl.fail:-2
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] tarpitting connect for 0s
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] connect: -2, history: 127, total_connects: 264, neighbors: -4184, pass:fcrdns.fcrdns, fail:fcrdns.fail, dnsbl.fail, neighbors(-4184), skip:penalty
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] tarpit connect end
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [access] pass:connect:any, helo:any
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [spf] identity=helo ip=50.159.99.59 domain="openssl.client.net" mfrom=<postmaster@openssl.client.net> result=None
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [spf] scope: helo, result: None, domain: openssl.client.net
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] applied results.karma.neighbors:-1
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] applied results.karma.neighbors:-1
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] applied results.helo.checks.fail:-1
Apr 12 12:13:35 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] tarpitting ehlo for 2s
Apr 12 12:13:37 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] tarpit ehlo end
Apr 12 12:13:38 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [core] client [50.159.99.59] connection error: Error: 34410095616:error:1408A0C1:SSL routines:SSL3_GET_CLIENT_HELLO:no shared cipher:../deps/openssl/openssl/ssl/s3_srvr.c:1360:
Apr 12 12:13:38 node haraka[39750]: [INFO] [95E05881-08D9-4253-9D96-452D86B52E55] [karma] connect: -5, history: 126, total_connects: 264, neighbors: -4184, msg:good enough hist, pass:fcrdns.fcrdns, fail:fcrdns.fail, dnsbl.fail, neighbors(-4184), karma.neighbors, karma.neighbors, helo.checks.fail, skip:penalty
Apr 12 12:13:38 node haraka[39750]: [NOTICE] [95E05881-08D9-4253-9D96-452D86B52E55] [core] disconnect ip=50.159.99.59 rdns="c-50-159-99-59.hsd1.wa.comcast.net" helo="openssl.client.net" relay=N early=N esmtp=Y tls=N pipe=N txns=0 rcpts=0/0/0 msgs=0/0/0 bytes=0 lr="" time=3.57
````

**Comment by celesteking** [comment]
Did you wait enough time?  What's tls plugin timeout value?

```
 $ openssl s_client -connect  localhost:2525  -starttls smtp  -cipher ECDHE-ECDSA-AES128-SHA
CONNECTED(00000005)
140373926856360:error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure:s23_lib.c:184:

[DEBUG] [-] [core] plugin tls timeout is: 30s
...
[NOTICE] [D499022D-5F23-46A5-B1F8-91DE2E0A9F05] [core] disconnect ip=127.0.0.1 rdns="river.home" helo="openssl.client.net" relay=N early=N esmtp=Y tls=N pipe=N txns=0 rcpts=0/0/0 msgs=0/0/0 bytes=0 lr="" time=0.045
[ERROR] [D499022D-5F23-46A5-B1F8-91DE2E0A9F05] [tls] timeout
```

**Comment by smfreegard** [comment]
Fixed in #922

**Comment by celesteking** [comment]
thanks sweeties
